controllers/awsmachine_controller.go

```diff
@@ -411,7 +412,7 @@ func (r *AWSMachineReconciler) reconcileNormal(_ context.Context, machineScope *
 	}
 
 	// Make sure Spec.ProviderID is always set.
-	machineScope.SetProviderID(fmt.Sprintf("aws:////%s", instance.ID))
+	machineScope.SetProviderID(fmt.Sprintf("aws:///%s/%s", aws.StringValue(machineScope.AWSMachine.Spec.FailureDomain), instance.ID))
```

Comment:
This still needs to be addressed https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1730#discussion_r428048160